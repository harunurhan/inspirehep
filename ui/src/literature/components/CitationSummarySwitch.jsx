import React from 'react';
import { Icon, Switch, Tooltip } from 'antd';

const CHART_ICON = <Icon type="bar-chart" />;

function CitationSummarySwitch(props) {
  const { checked } = props;

  const actionName = checked ? 'Hide' : 'Show';
  const tooltipHelp = `${actionName} Citation Summary`;
  return (
    <Tooltip title={tooltipHelp}>
      <Switch
        checkedChildren={CHART_ICON}
        unCheckedChildren={CHART_ICON}
        {...props}
      />
    </Tooltip>
  );
}

export default CitationSummarySwitch;
