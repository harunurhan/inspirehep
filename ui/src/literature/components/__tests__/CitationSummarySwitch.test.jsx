import React from 'react';
import { shallow } from 'enzyme';
import { Switch } from 'antd';

import CitationSummarySwitch from '../CitationSummarySwitch';

describe('CitationSummarySwitch', () => {
  it('renders checked', () => {
    const wrapper = shallow(
      <CitationSummarySwitch checked onChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders unchecked', () => {
    const wrapper = shallow(
      <CitationSummarySwitch checked={false} onChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onChange on switch change', () => {
    const onChange = jest.fn();

    const wrapper = shallow(
      <CitationSummarySwitch checked onChange={onChange} />
    );

    const onSwitchChange = wrapper.find(Switch).prop('onChange');
    onSwitchChange(false);

    expect(onChange).toHaveBeenCalledWith(false);
  });
});
