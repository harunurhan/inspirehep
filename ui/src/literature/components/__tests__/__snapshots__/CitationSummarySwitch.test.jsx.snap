// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitationSummarySwitch renders checked 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="top"
  title="Hide Citation Summary"
  transitionName="zoom-big-fast"
>
  <Switch
    checked={true}
    checkedChildren={
      <Icon
        type="bar-chart"
      />
    }
    onChange={[MockFunction]}
    unCheckedChildren={
      <Icon
        type="bar-chart"
      />
    }
  />
</Tooltip>
`;

exports[`CitationSummarySwitch renders unchecked 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="top"
  title="Show Citation Summary"
  transitionName="zoom-big-fast"
>
  <Switch
    checked={false}
    checkedChildren={
      <Icon
        type="bar-chart"
      />
    }
    onChange={[MockFunction]}
    unCheckedChildren={
      <Icon
        type="bar-chart"
      />
    }
  />
</Tooltip>
`;
